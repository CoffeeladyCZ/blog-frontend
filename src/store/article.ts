import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { Article } from '../types/Articles';

interface ArticleState {
  listArticles: Article[];
}

const initialState: ArticleState = {
  listArticles: []
};

export const articleSlice = createSlice({
  name: 'listArticles',
  initialState,
  reducers: {
    setListArticles: (state, action: PayloadAction<Article[]>) => {
      state.listArticles = action.payload;
    }
  }
});

export const { setListArticles } = articleSlice.actions;
export default articleSlice.reducer;
