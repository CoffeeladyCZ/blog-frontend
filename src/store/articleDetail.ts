import { ArticleDetailTypes } from './../types/Articles';
import { createSlice, PayloadAction } from '@reduxjs/toolkit';

interface ArticleDetailState {
  articleDetail: ArticleDetailTypes | null;
}

const initialArticleDetailState: ArticleDetailState = {
  articleDetail: null
};

export const articleDetailSlice = createSlice({
  name: 'articleDetail',
  initialState: initialArticleDetailState,
  reducers: {
    setArticleDetail: (state, action: PayloadAction<ArticleDetailTypes | null>) => {
      state.articleDetail = action.payload;
    }
  }
});

export const { setArticleDetail } = articleDetailSlice.actions;

export const articleDetailReducer = articleDetailSlice.reducer;
