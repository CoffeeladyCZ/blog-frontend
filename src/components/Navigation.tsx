import React, { useState, useEffect } from 'react';
import { NavLink, useNavigate } from 'react-router-dom';
import Cookies from 'js-cookie';
import { useDispatch, useSelector } from 'react-redux';
import { useTranslation } from 'react-i18next';

import {
  AppBar,
  Box,
  Button,
  Container,
  CssBaseline,
  Divider,
  Drawer,
  IconButton,
  List,
  ListItem,
  ListItemButton,
  ListItemText,
  Toolbar
} from '@mui/material';
import { Login } from '@mui/icons-material';
import MenuIcon from '@mui/icons-material/Menu';
import { styled } from '@mui/material/styles';

import { RootState } from '../store/store';
import { setLogin } from '../store/login';
import { StyledNavLink, StyledNavButton } from '../styled/styled';

import logo from '../assets/logo.svg';

import Notification from './Notification';
import LanguageSwitch from './LanguageSwitch';

type LoginSectionType = {
  onUserAlert: (item: boolean) => void;
  onCloseMenu: (item: boolean) => void;
};

const StyledAppBar = styled(AppBar)`
  box-shadow: none;
`;

const StyledNavBox = styled(Box)`
  flex-grow: 1;
`;

const StyledLoginBox = styled(Box)`
  flex-grow: 0;
`;

const StyledLanguageBox = styled(Box)`
  flex-grow: 0;
`;

const LoginSection: React.FC<LoginSectionType> = ({ onUserAlert, onCloseMenu }) => {
  const dispatch = useDispatch();
  const login = useSelector((state: RootState) => state.login.login);
  const navigate = useNavigate();
  const { t } = useTranslation();

  const loginPages = [
    {
      value: 'my-article',
      name: t('navigation.myArticles'),
      link: '/articles',
      testId: 'navMyArticles'
    },
    {
      value: 'new-article',
      name: t('navigation.newArticle'),
      link: '/article/new',
      testId: 'navNewArticles'
    },
    { value: 'logout', name: t('logout'), link: '', testId: 'logout' }
  ];

  useEffect(() => {
    const token = Cookies.get('token');
    const isLogin: boolean = token ? true : false;
    dispatch(setLogin(isLogin));
  }, [dispatch]);

  const handleClick = (value: string) => {
    if (value === 'logout') {
      logoutUser(false);
    } else {
      onCloseMenu(true);
    }
  };

  /** Log out user from application */
  const logoutUser = (disconnected: boolean) => {
    dispatch(setLogin(false));
    Cookies.remove('token');
    localStorage.removeItem('loginTime');
    clearInterval(expirationCheckInterval);

    //props drilling to parent component for display notification
    onUserAlert(true);

    // if an hour has passed, the user is logged out
    if (disconnected) {
      navigate('/disconnected');
      return;
    }
    navigate('/recent-articles');
  };

  const checkLoginExpiration = () => {
    const loginTimeFromStorage = localStorage.getItem('loginTime');

    if (loginTimeFromStorage !== null) {
      const currentTime: Date = new Date();
      const loginTime = new Date(loginTimeFromStorage);
      const timeDifferenceMinutes = (currentTime.getTime() - loginTime.getTime()) / 1000;
      if (timeDifferenceMinutes >= 3600) {
        logoutUser(true);
      }
    }
  };

  const expirationCheckInterval = setInterval(checkLoginExpiration, 600000); // 10 minutes

  if (!login) {
    return (
      <NavLink to="/login">
        <Button data-testid="navLoginButton" startIcon={<Login />}>
          {t('login')}
        </Button>
      </NavLink>
    );
  } else {
    return (
      <>
        <Box
          data-testid="loginSection"
          sx={{
            display: { xs: 'none', sm: 'flex' }
          }}>
          {loginPages.map((page) => (
            <StyledNavButton
              key={page.value}
              onClick={() => handleClick(page.value)}
              sx={{ my: 2, display: 'block' }}>
              <StyledNavLink key={page.value} to={page.link} data-testid={page.testId}>
                {page.name}
              </StyledNavLink>
            </StyledNavButton>
          ))}
        </Box>
        <List
          sx={{
            display: { xs: 'block', sm: 'none' },
            '& .MuiDrawer-paper': { boxSizing: 'border-box' },
            padding: 0
          }}>
          {loginPages.map((item) => (
            <ListItem key={item.value} disablePadding>
              <ListItemButton sx={{ textAlign: 'center' }}>
                <ListItemText primary={item.name} onClick={() => handleClick(item.value)} />
              </ListItemButton>
            </ListItem>
          ))}
        </List>
      </>
    );
  }
};

const Navigation: React.FC = () => {
  const [showSuccessAlert, setShowSuccessAlert] = useState<boolean>(false);
  const [mobileOpen, setMobileOpen] = useState(false);

  const setUserAlert: (item: boolean) => void = (item) => {
    setShowSuccessAlert(item);
  };
  const { t } = useTranslation();
  const pages = [
    {
      value: 'recent-article',
      name: t('navigation.recentArticles'),
      link: '/recent-articles',
      testId: 'navRecentArticles'
    }
  ];

  const handleDrawerToggle = () => {
    setMobileOpen((prevState) => !prevState);
  };

  const drawerWidth = 240;
  const container = window !== undefined ? () => document.body : undefined;

  const drawer = (
    <Box onClick={handleDrawerToggle} sx={{ textAlign: 'center' }}>
      <img src={logo} alt="logo" id="logo" />
      <Divider />
      <List>
        {pages.map((item) => (
          <ListItem key={item.value} disablePadding>
            <ListItemButton sx={{ textAlign: 'center' }}>
              <ListItemText primary={item.name} />
            </ListItemButton>
          </ListItem>
        ))}
        <LoginSection onUserAlert={setUserAlert} onCloseMenu={handleDrawerToggle} />
      </List>
    </Box>
  );

  return (
    <Box display="flex">
      <CssBaseline />
      <StyledAppBar color="secondary">
        {showSuccessAlert && (
          <Notification
            severity="info"
            message="User has been logged out."
            open={showSuccessAlert}
            onClose={() => setShowSuccessAlert(false)}
          />
        )}
        <Container max-width="xl">
          <Toolbar>
            <IconButton
              color="inherit"
              aria-label="open drawer"
              edge="start"
              onClick={handleDrawerToggle}
              sx={{ mr: 2, display: { sm: 'none' } }}>
              <MenuIcon />
            </IconButton>
            <IconButton size="large" edge="start" color="inherit" aria-label="menu">
              <img src={logo} alt="logo" id="logo" />
            </IconButton>
            <StyledNavBox sx={{ display: { xs: 'none', sm: 'block' } }}>
              {pages.map((page) => (
                <StyledNavButton key={page.value}>
                  <StyledNavLink to={page.link} data-testid={page.testId}>
                    {page.name}
                  </StyledNavLink>
                </StyledNavButton>
              ))}
            </StyledNavBox>
            <StyledLoginBox sx={{ display: { xs: 'none', sm: 'block' } }}>
              <LoginSection onUserAlert={setUserAlert} onCloseMenu={handleDrawerToggle} />
            </StyledLoginBox>
            <StyledLanguageBox>
              <LanguageSwitch />
            </StyledLanguageBox>
            
          </Toolbar>
        </Container>
      </StyledAppBar>
      <nav>
        <Drawer
          container={container}
          variant="temporary"
          open={mobileOpen}
          onClose={handleDrawerToggle}
          ModalProps={{
            keepMounted: true
          }}
          sx={{
            display: { xs: 'block', sm: 'none' },
            '& .MuiDrawer-paper': { boxSizing: 'border-box', width: drawerWidth }
          }}>
          {drawer}
        </Drawer>
      </nav>
    </Box>
  );
};

export default Navigation;
