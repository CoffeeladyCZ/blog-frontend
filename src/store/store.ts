import { configureStore } from '@reduxjs/toolkit';
import { articleSlice } from './article';
import { articleDetailSlice } from './articleDetail';
import { loginSlice } from './login';

const store = configureStore({
  reducer: {
    articleList: articleSlice.reducer,
    articleDetail: articleDetailSlice.reducer,
    login: loginSlice.reducer
  }
});

export type AppStore = ReturnType<typeof store.getState>;
export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export default store;
