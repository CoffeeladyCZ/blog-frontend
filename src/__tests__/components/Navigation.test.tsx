import { render, screen, fireEvent, act } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import Cookies from 'js-cookie';

import { setLogin } from '../../store/login';

import Navigation from '../../components/Navigation';

describe('Navigation without login', () => {
  const initialState = {
    login: {
      login: false
    }
  };
  const mockStore = configureStore([]);
  const store = mockStore(initialState);

  test('should snapshot with data', () => {
    const container = render(
      <Provider store={store}>
        <MemoryRouter initialEntries={['/login']}>
          <Navigation />
        </MemoryRouter>
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });

  test('component is succesfully render - without login', () => {
    render(
      <Provider store={store}>
        <MemoryRouter initialEntries={['/login']}>
          <Navigation />
        </MemoryRouter>
      </Provider>
    );

    expect(screen.getByText('Recent Articles')).toBeInTheDocument();
  });
});

describe('Navigation with login', () => {
  const initialState = {
    login: {
      login: true
    }
  };
  const mockStore = configureStore([]);
  const store = mockStore(initialState);

  test('should snapshot with data', () => {
    const container = render(
      <Provider store={store}>
        <MemoryRouter initialEntries={['/login']}>
          <Navigation />
        </MemoryRouter>
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });

  test('component is succesfully render - with login', () => {
    render(
      <Provider store={store}>
        <MemoryRouter initialEntries={['/login']}>
          <Navigation />
        </MemoryRouter>
      </Provider>
    );

    expect(screen.getByText('My Articles')).toBeInTheDocument();
    expect(screen.getByText('New Article')).toBeInTheDocument();
  });

  test('logoutUser correctly dispatches actions and updates state', () => {
    Cookies.remove = jest.fn();
    store.dispatch = jest.fn();

    render(
      <Provider store={store}>
        <MemoryRouter>
          <Navigation />
        </MemoryRouter>
      </Provider>
    );

    const menuButton = screen.getByTestId('menuButton');
    fireEvent.click(menuButton);
    const submitButton = screen.getByTestId('logout');
    act(() => {
      fireEvent.click(submitButton);
    });

    expect(store.dispatch).toHaveBeenCalledWith(setLogin(false));
    expect(Cookies.remove).toHaveBeenCalledWith('token');
  });
});
