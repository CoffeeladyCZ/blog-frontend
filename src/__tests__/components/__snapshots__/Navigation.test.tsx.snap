// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation with login should snapshot with data 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorSecondary MuiAppBar-positionRelative css-15o3gvq-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg css-1kgzgdg-MuiContainer-root"
        >
          <div
            class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-dense css-f59th8-MuiToolbar-root"
          >
            <a
              class=""
              href="/"
            >
              <button
                aria-label="menu"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge css-184mzhy-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <img
                  alt="logo"
                  id="logo"
                  src="logo.svg"
                />
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </a>
            <div
              class="MuiBox-root css-nv9fs0"
            >
              <a
                class="css-11zjqnh"
                data-testid="navRecentArticles"
                href="/recent-articles"
              >
                Recent Articles
              </a>
            </div>
            <div
              class="MuiBox-root css-nv9fs0"
            >
              <div>
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="button"
                >
                  Čeština
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
            <div
              class="MuiBox-root css-1h0u4nz"
              data-testid="loginSection"
            >
              <a
                class="css-11zjqnh"
                data-testid="navMyArticles"
                href="/articles/"
              >
                My Articles
              </a>
              <a
                class="css-11zjqnh"
                data-testid="navNewArticle"
                href="/article/new"
              >
                New Article
              </a>
              <button
                aria-label="Open menu"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                data-mui-internal-clone-element="true"
                data-testid="menuButton"
                tabindex="0"
                type="button"
              >
                <div
                  class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
                >
                  <img
                    alt="avatar"
                    class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                    srcset="../assets/logo_A.jpg"
                  />
                </div>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </header>
    </div>
    <div
      aria-hidden="true"
      class="MuiMenu-root MuiPopover-root MuiModal-root MuiModal-hidden css-o9p8ou-MuiModal-root-MuiPopover-root-MuiMenu-root"
      id="userMenu"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root MuiBackdrop-invisible MuiModal-backdrop css-g3hgs1-MuiBackdrop-root-MuiModal-backdrop"
        style="opacity: 0; visibility: hidden;"
      />
      <div
        data-testid="sentinelStart"
        tabindex="-1"
      />
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 MuiMenu-paper MuiPopover-paper MuiMenu-paper css-3dzjca-MuiPaper-root-MuiPopover-paper-MuiMenu-paper"
        style="opacity: 0; transform: scale(0.75, 0.5625); visibility: hidden;"
        tabindex="-1"
      >
        <ul
          class="MuiList-root MuiList-padding MuiMenu-list css-6hp17o-MuiList-root-MuiMenu-list"
          role="menu"
          tabindex="-1"
        >
          <li
            class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
            data-testid="logout"
            role="menuitem"
            tabindex="0"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1uwgr7b-MuiTypography-root"
            >
              Log Out
            </p>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </li>
        </ul>
      </div>
      <div
        data-testid="sentinelEnd"
        tabindex="-1"
      />
    </div>
  </body>,
  "container": <div>
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorSecondary MuiAppBar-positionRelative css-15o3gvq-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-1kgzgdg-MuiContainer-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-dense css-f59th8-MuiToolbar-root"
        >
          <a
            class=""
            href="/"
          >
            <button
              aria-label="menu"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge css-184mzhy-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <img
                alt="logo"
                id="logo"
                src="logo.svg"
              />
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </a>
          <div
            class="MuiBox-root css-nv9fs0"
          >
            <a
              class="css-11zjqnh"
              data-testid="navRecentArticles"
              href="/recent-articles"
            >
              Recent Articles
            </a>
          </div>
          <div
            class="MuiBox-root css-nv9fs0"
          >
            <div>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Čeština
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          <div
            class="MuiBox-root css-1h0u4nz"
            data-testid="loginSection"
          >
            <a
              class="css-11zjqnh"
              data-testid="navMyArticles"
              href="/articles/"
            >
              My Articles
            </a>
            <a
              class="css-11zjqnh"
              data-testid="navNewArticle"
              href="/article/new"
            >
              New Article
            </a>
            <button
              aria-label="Open menu"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              data-mui-internal-clone-element="true"
              data-testid="menuButton"
              tabindex="0"
              type="button"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
              >
                <img
                  alt="avatar"
                  class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                  srcset="../assets/logo_A.jpg"
                />
              </div>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Navigation without login should snapshot with data 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorSecondary MuiAppBar-positionRelative css-15o3gvq-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg css-1kgzgdg-MuiContainer-root"
        >
          <div
            class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-dense css-f59th8-MuiToolbar-root"
          >
            <a
              class=""
              href="/"
            >
              <button
                aria-label="menu"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge css-184mzhy-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <img
                  alt="logo"
                  id="logo"
                  src="logo.svg"
                />
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </a>
            <div
              class="MuiBox-root css-nv9fs0"
            >
              <a
                class="css-11zjqnh"
                data-testid="navRecentArticles"
                href="/recent-articles"
              >
                Recent Articles
              </a>
            </div>
            <div
              class="MuiBox-root css-nv9fs0"
            >
              <div>
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="button"
                >
                  Čeština
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
            <a
              aria-current="page"
              class="active"
              href="/login"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Log In
                <span
                  class="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="ArrowForwardIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="m12 4-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </a>
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorSecondary MuiAppBar-positionRelative css-15o3gvq-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-1kgzgdg-MuiContainer-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-dense css-f59th8-MuiToolbar-root"
        >
          <a
            class=""
            href="/"
          >
            <button
              aria-label="menu"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge css-184mzhy-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <img
                alt="logo"
                id="logo"
                src="logo.svg"
              />
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </a>
          <div
            class="MuiBox-root css-nv9fs0"
          >
            <a
              class="css-11zjqnh"
              data-testid="navRecentArticles"
              href="/recent-articles"
            >
              Recent Articles
            </a>
          </div>
          <div
            class="MuiBox-root css-nv9fs0"
          >
            <div>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Čeština
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          <a
            aria-current="page"
            class="active"
            href="/login"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Log In
              <span
                class="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ArrowForwardIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="m12 4-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </a>
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
